site/sfguides/src/native-app-chairlift/native-app-chairlift.md

```diff
@@ -0,0 +1,632 @@
+        select id, uuid, min_range, max_range
+          from sensors s join sensor_types sr
+                 on s.sensor_type_id = sr.id;
+    begin
+      --
+      -- starting_ts is the time of the last sensor reading we wrong or, if no
+      -- readings are available, 10 minutes in the past.
+      --
+      select coalesce(max(reading_time), dateadd(second, -30*20, current_timestamp()))
+               into :starting_ts
+        from sensor_readings;
+
+      --
+      -- produce one row for every thirty seconds from our starting time to now
+      --
+      rows_to_produce := datediff(second, starting_ts, current_timestamp()) / 30;
+
+      for sensor in sensors_cursor do
+        let sensor_uuid varchar default sensor.uuid;
+        let min_range integer default sensor.min_range;
+        let max_range integer default sensor.max_range;
+  
+        insert into sensor_readings(sensor_uuid, reading_time, reading)
+          select
+              :sensor_uuid,
+              dateadd(second, row_id * 30, :starting_ts),
+              case
+                when rand_value < 10 then
+                  :min_range - abs(hash(uuid)) % 10
+                when rand_value > 90 then
+                  :max_range + abs(hash(uuid)) % 10
+                else
+                  :min_range + abs(hash(uuid)) % (:max_range - :min_range)
+              end case
+          from ( 
+              select seq4() + 1            as row_id,
+                     uuid_string()         as uuid,
+                     abs(hash(uuid)) % 100 as rand_value
+                from table(generator(rowcount => :rows_to_produce)));
+      end for;
+
+      update sensors
+         set last_reading = r.reading
+        from sensors as s2, sensor_readings as r
+       where s2.uuid = sensors.uuid
+         and r.sensor_uuid = s2.uuid
+         and r.reading_time = 
+              (select max(reading_time)  
+                 from sensor_readings r2
+                where r2.sensor_uuid = s2.uuid);
+    end;
+  $$
+;
+
+-- Task to call the stored procedure to update the readings table every minute
+create or replace task populate_reading_every_minute
+    warehouse = chairlift_wh
+    schedule = '1 minute'
+as
+    call populate_reading();
+
+-- Get some initial data in the readings table
+call populate_reading();
+
+-- If you would like the data to be populated on a schedule, you can run:
+-- alter task chairlift_consumer_data.data.populate_reading_every_minute resume;
+
+-- To stop:
+-- alter task chairlift_consumer_data.data.populate_reading_every_minute suspend;
+```
+
+<!-- ------------------------ -->
+## Create application package
+Duration: 2
+
+With the environment created, we can now create the application package for the app. You'll run a script that creates this package and does a few key things:
+
+* Creates the application package for the native app
+
+* Marks that the native app will make use of certain data in the consumer's account
```

Comment:
I don't think this is true. This is the responsibility of the manifest; everything here is provider-side.

Reply:
See file in Slack.